datafusion/core/src/physical_plan/projection.rs

```diff
@@ -221,6 +232,10 @@ impl ExecutionPlan for ProjectionExec {
         )?))
     }
 
+    fn benefits_from_input_partitioning(&self) -> bool {
+        self.would_benefit
```

Comment:
I would suggest to put the code here and compute it on demand.

Reply:
Agreed 👍